core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -242,17 +251,37 @@ class ResourceProfile(
 
 object ResourceProfile extends Logging {
   // task resources
+  /**
+   * built-in task resource: cpus
+   */
   val CPUS = "cpus"
   // Executor resources
   // Make sure add new executor resource in below allSupportedExecutorResources
+  /**
+   * built-in executor resource: cores
+   */
   val CORES = "cores"
+  /**
+   * built-in executor resource: cores
+   */
   val MEMORY = "memory"
+  /**
+   * built-in executor resource: offHeap
+   */
   val OFFHEAP_MEM = "offHeap"
+  /**
+   * built-in executor resource: memoryOverhead
+   */
   val OVERHEAD_MEM = "memoryOverhead"
+  /**
+   * built-in executor resource: pyspark.memory
+   */
   val PYSPARK_MEM = "pyspark.memory"
 
-  // all supported spark executor resources (minus the custom resources like GPUs/FPGAs)
-  val allSupportedExecutorResources = Seq(CORES, MEMORY, OVERHEAD_MEM, PYSPARK_MEM, OFFHEAP_MEM)
+  /**
+   * All supported Spark built-in executor resources, custom resources like GPUs/FPGAs are excluded.
+   */
+  val allSupportedExecutorResources = Array(CORES, MEMORY, OVERHEAD_MEM, PYSPARK_MEM, OFFHEAP_MEM)
```

Comment:
Why did we change this to Array ?
Seq is immutable, while Array allows updates (accidental or otherwise).
Can we revert this ?

Reply:
Oh @mridulm I think this is actually for Java users (to hide Scala specific classes to end Java users). I believe we have done this in places such as `Dataset`.

Reply:
Make sense to me. We changed to `Array` for Java-friendly purpose. 

I have added the java specific func to address the immutable concern: 12f5481

But this also introduces a new API, @tgravescs  are you ok with it?